### DISCUSSION TITLE
New predict() behaviour and syntax?

### DISCUSSION THREAD

**Discussion by DominiqueMakowski** [TuringLang/Turing.jl#2387] [Q&A]
It's me again, just wanted some clarification, as it seems like recently the behaviour of `predict()` changed in the recent updates?
Previously if you ran `predict(model([vector_of_missing]), chains)` and it would return an array of the size of the vector of missing it's been conditioned on with the draws, but now the same function returns also all the parameters (?) 

What is the correct way moving forward of generating predictions on the outcome of a model? 

If the current `predict()` output is here to stay, how can I easily filter out the draws corresponding to my outcome? In my case, I'm interested in predicting a variable called response time `rt`. And the chains output has the following names:

```jl
julia>     names(pred)
3033-element Vector{Symbol}:
 :τ_intercept_random_sd
 :τ_isi1_random_sd
 :τ_isi2_random_sd
 Symbol("τ_intercept_random[1]")
 Symbol("τ_intercept_random[2]")
 Symbol("τ_intercept_random[3]")
 Symbol("τ_intercept_random[4]")
 ⋮
 Symbol("rt[2950]")
 Symbol("rt[2951]")
 Symbol("rt[2952]")
 Symbol("rt[2953]")
 Symbol("rt[2954]")
 Symbol("rt[2955]")
```

How can I do something like `pred[parameters_starting_with_rt]` the most Turingy/julian way?

**Comment by torfjelde** [comment]
Hmm, I think this is a bug and not a feature 😕 I'll investigate 🕵️
